File: HackerRank/30 Days of Code/Day15_LinkedList.cs
// Day 15: Linked List
// https://www.hackerrank.com/challenges/30-linked-list/problem

using System;
class Node
{
	public int data;
	public Node next;
    public Node(int d){
        data=d;
        next=null;
    }
		
}
class Solution {

	public static  Node insert(Node head,int data)
	{
        
      if (head != null){
          Node cur = head;
          while(cur.next != null)
            cur = cur.next;
          cur.next = new Node(data);
      }
      else
          head = new Node(data);
      return head;
    }

	public static void display(Node head)
	{
		Node start=head;
		while(start!=null)
		{
			Console.Write(start.data+" ");
			start=start.next;
		}
	}
    static void Main(String[] args) {
	
		Node head=null;
        int T=Int32.Parse(Console.ReadLine());
        while(T-->0){
            int data=Int32.Parse(Console.ReadLine());
            head=insert(head,data);
        }
		display(head);
	}
}
		  
// 연결 리스트 구현
// C#은 객체와 null로 구현

File: HackerRank/30 Days of Code/Day15_LinkedList.java
// Day 15: Linked List
// https://www.hackerrank.com/challenges/30-linked-list/problem

import java.io.*;
import java.util.*;

class Node {
	int data;
	Node next;
	Node(int d) {
        data = d;
        next = null;
    }
}

class Solution {

    public static  Node insert(Node head,int data) {
        if (head != null){
            Node cur = head;
            while(cur.next != null)
                cur = cur.next;
            cur.next = new Node(data);
        }
        else
            head = new Node(data);
        return head;
    }

	public static void display(Node head) {
        Node start = head;
        while(start != null) {
            System.out.print(start.data + " ");
            start = start.next;
        }
    }

    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        Node head = null;
        int N = sc.nextInt();

        while(N-- > 0) {
            int ele = sc.nextInt();
            head = insert(head,ele);
        }
        display(head);
        sc.close();
    }
}
		  
// 연결 리스트 구현
// Java는 객체와 null로 구현